library/src/main/java/org/apache/apex/malhar/lib/state/spillable/SpillableMapImpl.java

```diff
@@ -236,13 +237,27 @@ public void beginWindow(long windowId)
   @Override
   public void endWindow()
   {
+    boolean isTimeUnifiedStore = (store instanceof ManagedTimeUnifiedStateImpl);
```

Comment:
How would this increase the complexity? I think it would simplify the logic as this would need to be done just once in setup().